client/my-sites/checkout/composite-checkout/composite-checkout.js

```diff
@@ -624,6 +628,30 @@ function useCountryList( overrideCountryList ) {
 	return countriesList;
 }
 
+function useRecordCheckoutLoaded(
+	recordEvent,
+	isApplePayAvailable,
+	responseCart,
+	storedCards,
+	isLoadingStoredCards,
+	product
+) {
+	const hasRecordedCheckoutLoad = useRef( false );
+	if ( ! isLoadingStoredCards && ! hasRecordedCheckoutLoad.current ) {
+		debug( 'composite checkout has loaded' );
+		recordEvent( {
+			type: 'CHECKOUT_LOADED',
+			payload: {
+				saved_cards: storedCards.length,
+				apple_pay_available: isApplePayAvailable,
```

Comment:
Since `isApplePayAvailable` is also an async property, you should probably add `isApplePayLoading` to the `if` condition. Alternatively, just use `arePaymentMethodsLoading` which includes a lot of things, including the cart, which you use below for `is_renewal`.

The only downside I can think of there is that it will never be `false` if the cart is empty or if there's an error loading the cart, but maybe that doesn't matter? Anyway, worth looking at those conditions.